app/controllers/lead_providers/partnership_csv_uploads_controller.rb

```diff
@@ -7,26 +7,37 @@ def new
     end
 
     def create
-      # TODO: do we want to catch empty session[:delivery_partner_id] ?
       if params[:partnership_csv_upload].blank?
         @partnership_csv_upload = PartnershipCsvUpload.new
         @partnership_csv_upload.errors[:base] << "Please select a CSV file to upload."
         render :new and return
       end
 
-      @partnership_csv_upload = PartnershipCsvUpload.new(upload_params.merge(
-                                                           lead_provider_id: current_user.lead_provider_profile.lead_provider.id,
-                                                           delivery_partner_id: session[:delivery_partner_id],
-                                                         ))
+      @partnership_csv_upload = PartnershipCsvUpload.new(
+        upload_params.merge(
+          lead_provider_id: current_user.lead_provider_profile.lead_provider.id,
+          delivery_partner_id: session[:delivery_partner_id],
+        ),
+      )
 
       if @partnership_csv_upload.save
-        redirect_to error_page_lead_providers_report_schools_partnership_csv_uploads_path
+        session[:partnership_csv_upload_id] = @partnership_csv_upload.id
+        if @partnership_csv_upload.invalid_schools.empty?
+          redirect_to lead_providers_report_schools_confirm_schools_path
+        else
+          redirect_to errors_lead_providers_report_schools_partnership_csv_uploads_path
+        end
       else
         render :new
       end
     end
 
-    def error_page; end
+    def errors
+      partnership_csv_upload = PartnershipCsvUpload.find(session[:partnership_csv_upload_id])
```

Comment:
I am not too thrilled about having multiple session parameters in the flow. It makes tracking when we delete them hard :/ Can we make them into a serializable form? Or have a single hash for this flow?

Reply:
The thing is, we really want to delete them at different times. We want to delete this one as soon as we've processed it, but keep the delivery partner ID for a while (through CSV reuploads etc.)